hydra-node/test/Hydra/API/ServerSpec.hs

```diff
@@ -267,9 +277,105 @@ spec = describe "ServerSpec" $
+                status `shouldBe` Aeson.String "Initializing"
+
+    it "displays correctly snapshotUtxo in a Greeting message" $
+      showLogsOnFailure $ \tracer ->
+        withFreePort $ \port -> do
+          withAPIServer @SimpleTx "127.0.0.1" port alice mockPersistence tracer noop $ \Server{sendOutput} -> do
+            generatedSnapshot :: Snapshot SimpleTx <- generate arbitrary
+
+            let snapShotConfirmedMessage =
+                  SnapshotConfirmed
+                    { headId = HeadId "some-head-id"
+                    , snapshot = generatedSnapshot
+                    , signatures = mempty
+                    }
+            let expectedUtxos =
+                  toJSON (Hydra.Snapshot.utxo $ Hydra.API.ServerOutput.snapshot snapShotConfirmedMessage)
+            sendOutput snapShotConfirmedMessage
+            withClient port "/?history=no" $ \conn -> do
+              status <- waitMatch 5 conn $ \v -> v ^? key "snapshotUtxo"
+              status `shouldBe` expectedUtxos
+
+            -- send another output related to changing the utxo set
+            headIsOpen <- HeadIsOpen @SimpleTx <$> generate arbitrary <*> generate arbitrary
+            let expectedUtxos' = toJSON $ Hydra.API.ServerOutput.utxo headIsOpen
+            sendOutput headIsOpen
+            withClient port "/?history=no" $ \conn -> do
+              status <- waitMatch 5 conn $ \v -> v ^? key "snapshotUtxo"
+              status `shouldBe` expectedUtxos'
+
+    it "greets with correct snapshot utxo after a restart" $
+      showLogsOnFailure $ \tracer ->
+        withTempDir "api-server-snapshot-utxo" $ \persistenceDir ->
+          withFreePort $ \port -> do
+            apiPersistence <- createPersistenceIncremental $ persistenceDir <> "/server-output"
+            generatedSnapshot :: Snapshot SimpleTx <- generate arbitrary
+            let snapShotConfirmedMessage =
+                  SnapshotConfirmed
+                    { headId = HeadId "some-head-id"
+                    , snapshot = generatedSnapshot
+                    , signatures = mempty
+                    }
+            let expectedUtxos =
+                  toJSON (Hydra.Snapshot.utxo $ Hydra.API.ServerOutput.snapshot snapShotConfirmedMessage)
+
+            withAPIServer @SimpleTx "127.0.0.1" port alice apiPersistence tracer noop $ \Server{sendOutput} -> do
+              sendOutput snapShotConfirmedMessage
+
+              withClient port "/?history=no" $ \conn -> do
+                status <- waitMatch 5 conn $ \v -> v ^? key "snapshotUtxo"
+                status `shouldBe` expectedUtxos
+
+            -- expect the api server to load events from apiPersistence and project headStatus correctly
+            withAPIServer @SimpleTx "127.0.0.1" port alice apiPersistence tracer noop $ \_ -> do
+              withClient port "/?history=no" $ \conn -> do
+                status <- waitMatch 5 conn $ \v -> v ^? key "snapshotUtxo"
+                status `shouldBe` expectedUtxos
```

Comment:
The headStatus and snapshotUtxo tests can be combined easily. They are both about the Greetings message, so that also sounds quite right to merge them.

Also there is quite some duplication between the restart and not restart tests, whereas the first part of the restart tests is exactly the same as the non-restart tests.

DRY please :)